linkerd/app/src/env.rs

```diff
@@ -280,6 +283,9 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
 
     let hostname = strings.get(ENV_HOSTNAME);
 
+    let f = fs::read_to_string(LABELS_FILE_PATH);
+    let labels = convert_string_to_map(f.unwrap_or_default());
```

Comment:
nit: The `unwrap_or_default` here will pass an empty string to `convert_string_to_map` if we couldn't read labels. Why not just skip calling this function? We could do something like
```suggestion
    let labels = f.ok().map(convert_string_to_map);
```
and avoid calling the function if the labels are empty.